Assets/Scripts/Player/Movement/PlayerMovement.cs

```diff
@@ -1,36 +1,32 @@
 using System;
-using System.Collections;
-using FrameWork.Extensions;
 using FrameWork.GridSystem;
-using UnityEngine;
 
 namespace Player.Movement
 {
-    public class PlayerMovement : MonoBehaviour
-    { 
-        [SerializeField, Range(1,10)] private float movementSpeed;
-        
-        private int _enumLength;
-        private int _gridIndex;
-
-        private void Awake()=> _enumLength = Enum.GetNames(typeof(GridPoints)).Length;
-        
-        private IEnumerator MoveTowardsGridPoint()
+    public sealed class PlayerMovement : Framework.Movement
+    {
+        /// <summary>
+        /// Calls the coroutine to start moving the player from it's current position to a newly assigned one.
+        /// </summary>
+        public void StartMoving(int playerPoints)
         {
-            _gridIndex = (_gridIndex + 1) % _enumLength;
-            GridPoints nextpoint = (GridPoints) _gridIndex;
-
-            Vector3 newPos;
-            newPos = nextpoint.GetVector3();
-            while (transform.position != newPos)
+            PlayerPoints points = (PlayerPoints)playerPoints;
+            switch (points)
             {
-                transform.position = Vector3.MoveTowards(transform.position, newPos, movementSpeed * Time.deltaTime);
-                yield return null;
+                case PlayerPoints.POINT_A:
+                    StartCoroutine(MoveTowardsGridPoint(PlayerPoints.POINT_A));
+                    break;
+                case PlayerPoints.POINT_B:
+                    StartCoroutine(MoveTowardsGridPoint(PlayerPoints.POINT_B));
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
```

Comment:
```suggestion
        public void StartMoving(PlayerPoints targetPoint) => StartCoroutine(MoveTowardsGridPoint<PlayerPoints>(points));
```

*scuffede review*, vraag niet waarom.